annotations/forms.py

```diff
@@ -27,10 +27,19 @@ class RegistrationForm(forms.Form):
 	"""
 	Gives user form of signup and validates it.
 	"""
-	username = forms.RegexField(regex=r'^\w+$', widget=forms.TextInput(attrs=dict(required=True, max_length=30)), label=_("Username"), error_messages={ 'invalid': _("This value must contain only letters, numbers and underscores.") })
+	full_name = forms.CharField(required=True, max_length=30, label=_("Full name"))
+	username = forms.RegexField(regex=r'^\w+$', widget=forms.TextInput(attrs=dict(required=True, max_length=30)),
+                                label=_("Username"),
+                                error_messages={ 'invalid': _("This value must contain only letters, "
+                                                              "numbers and underscores.") })
 	email = forms.EmailField(widget=forms.TextInput(attrs=dict(required=True, max_length=30)), label=_("Email address"))
-	password1 = forms.CharField(widget=forms.PasswordInput(attrs=dict(required=True, max_length=30, render_value=False)), label=_("Password"))
-	password2 = forms.CharField(widget=forms.PasswordInput(attrs=dict(required=True, max_length=30, render_value=False)), label=_("Password (again)"))
+	password1 = forms.CharField(widget=forms.PasswordInput(attrs=dict(required=True, max_length=30,
+                                                                      render_value=False)), label=_("Password"))
+	password2 = forms.CharField(widget=forms.PasswordInput(attrs=dict(required=True, max_length=30,
+                                                                      render_value=False)), label=_("Password (again)"))
+	affiliation = forms.CharField(required=True, max_length=30, label=_("Location"))
+	location = forms.CharField(required=False, max_length=30, label=_("Affliation"))
```

Comment:
Yes please :-) 

"When a user registers, full name, affiliation, and **location** should be mandatory fields"